### PR TITLE
Add uint64_t and int64_t types

### PR DESCRIPTION
**Pull request by mandolaerik**
- Repair test
- Add uint64_t and int64_t types

### CODE REVIEW

**Comment by syssimics** [comment]
Verification [#13186013](https://teamcity01-ir.devtools.intel.com/viewLog.html?buildId=13186013): pass

File: py/dml/types.py [hunk 1]
```diff
@@ -610,6 +612,29 @@ def clone(self):
...
+    `uint64_t *` arg. We find `uint64` a generally more useful type
+    primarily because it works better with format strings; "%lld" works
+    for `uint64` on both linux64 and win64.
+    '''
+    __slots__ = ()
+    def __init__(self, signed, const=False):
+        IntegerType.__init__(self, 64, signed, const=const)
+
+    def describe(self):
+        return self.declaration('').rstrip()
```
Comment (lwaern-intel) [inline py/dml/types.py:627]: ew. I'd vote for repetition instead (`return 'int64_t' if self.signed else 'uint64_t`), or, if you want to be particularly snazzy, leverage my "favorite" trick: `return 'u'*(not self.signed) + 'int64_t'`.

File: py/dml/types.py [hunk 2]
```diff
@@ -1271,6 +1296,13 @@ def compatible_types(type1, type2):
         ('size_t', TSize(False))]:
     typedefs[name] = typ
 
+if dml.globals.dml_version != (1, 2):
```
Comment (lwaern-intel) [inline py/dml/types.py:1299]: You shouldn't do this at top-level! `dml.globals.dml_version` can't be well-defined here, can it? Even if it is, that feels like that would _very_ easily break from any change we do re. when dml version is determined.

File: test/1.2/operators/T_sizeof.dml [hunk 3]
```diff
@@ -39,8 +39,8 @@ method test -> (bool ok) {
           (sizeoftype size_t == sizeoftype void *));
 
     // check that all sizeof invocations return a size_t
-    log "info": "%zd %zd %zd %zd", sizeof x, sizeof $x,
-        sizeoftype uint32, sizeoftype (t);
+    SIM_log_info(1, $dev.obj, 0, "%zd %zd %zd %zd", sizeof x, sizeof $x,
```
Comment (lwaern-intel) [inline test/1.2/operators/T_sizeof.dml:42]: What's this change about?

**Review by lwaern-intel — CHANGES_REQUESTED** [review]

File: test/1.2/operators/T_sizeof.dml (hunk 3 above)
Comment (mandolaerik) [reply test/1.2/operators/T_sizeof.dml:42]: Oh, I just found it on the way. IIRC, `log` adjust format strings to make qualifiers match args, so passing it as a `%zd` arg to `log` doesn't necessarily check size_t-ness, but passing it to SIM_log_info does. (format string conversions happen correctly in 1.4; I don't remember what limitations there are in 1.2 but this change is arguably an improvement)

File: py/dml/types.py (hunk 2 above)
Comment (mandolaerik) [reply py/dml/types.py:1299]: Oh, crap. Will fix it somehow.

File: test/1.2/operators/T_sizeof.dml (hunk 3 above)
Comment (lwaern-intel) [reply test/1.2/operators/T_sizeof.dml:42]: Alright.

File: doc/1.4/language.md [hunk 4]
```diff
@@ -1316,7 +1316,7 @@ The familiar integer types `char` and `int` are
 available as aliases for `int8` and `int32`,
 respectively. The C keywords `short`, `signed`, `unsigned`
 and `long` are reserved words in DML and not allowed in type
-declarations.  The types `size_t` and `uintptr_t` are
+declarations.  The types `uint64_t`, `int64_t`, `size_t` and `uintptr_t` are
 defined as in C.
```
Comment (lwaern-intel) [inline doc/1.4/language.md:1320]: This deserves some kind of clarifying disclaimer. How about:
> The types `uint64_t`, `int64_t`, `size_t` and `uintptr_t` are
defined as in C. (`uint64_t` and `int64_t` are guaranteed compatible with the types offered by `stdlib.h`, in contrast to `uint64` and `int64` which are defined by Simics.)

File: py/dml/types.py [hunk 5]
```diff
@@ -610,6 +612,33 @@ def clone(self):
...
+    primarily because it works better with format strings; "%lld" works
+    for `uint64` on both linux64 and win64.
+    '''
+    __slots__ = ()
+    def __init__(self, signed, const=False):
+        IntegerType.__init__(self, 64, signed, const=const)
+
+    def c_name(self):
+        name = 'int64_t' if self.signed else 'uint64_t'
+        return f'const {name}' if self.const else name
```
Comment (lwaern-intel) [inline py/dml/types.py:628]: Consider using `self.const_str()` like everything else (unless you're doing away with that function)

**Review by lwaern-intel — CHANGES_REQUESTED** [review]

File: py/dml/types.py (hunk 5 above)
Comment (mandolaerik) [reply py/dml/types.py:628]: done

File: doc/1.4/language.md (hunk 4 above)
Comment (mandolaerik) [reply doc/1.4/language.md]: changed

File: py/dml/types.py (hunk 1 above)
Comment (mandolaerik) [reply py/dml/types.py]: factored out a function

**Comment by syssimics** [comment]
Verification [#13268602](https://teamcity01-ir.devtools.intel.com/viewLog.html?buildId=13268602): pass

**Comment by syssimics** [comment]
Verification [#13268624](https://teamcity01-ir.devtools.intel.com/viewLog.html?buildId=13268624): pass

**Comment by syssimics** [comment]
Verification [#13268743](https://teamcity01-ir.devtools.intel.com/viewLog.html?buildId=13268743): pass

**Pull request merged**